modules/secrets-store-aws-secrets-manager.adoc

```diff
@@ -0,0 +1,245 @@
+----
+
+. Create a secret provider class to define your secrets store provider:
+
+.. Create a YAML file that defines the `SecretProviderClass` object:
++
+.Example `secret-provider-class-aws.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-aws-provider                   <1>
+  namespace: my-namespace                 <2>
+spec:
+  provider: aws                           <3>
+  parameters:                             <4>
+    objects: |
+      array:
+        - |
+          objectName: tls-cert
+          objectType: secret
+----
+<1> Specify the name for the provider secret provider class.
+<2> Specify the namespace for the provider secret provider class.
+<3> Specify the provider as `aws`.
+<4> Specify the provider-specific configuration parameters.
+
+.. Create the `SecretProviderClass` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret-provider-class-aws.yaml
+----
+
+. Create a deployment to use this secret provider class:
+
+.. Create a YAML file that defines the `Deployment` object:
++
+.Example `deployment.yaml`
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-aws-deployment                              <1>
+  namespace: my-namespace                              <2>
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: my-storage
+  template:
+    metadata:
+      labels:
+        app: my-storage
+    spec:
+      containers:
+      - name: busybox
+        image: k8s.gcr.io/e2e-test-images/busybox:1.29
+        command:
+          - "/bin/sleep"
+          - "10000"
+        volumeMounts:
+        - name: secrets-store-inline
+          mountPath: "/mnt/key-store"
+          readOnly: true
+      volumes:
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: "my-aws-provider" <3>
+----
+<1> Specify the name for the deployment.
+<2> Specify the namespace deployment.
+// TODO: Does this have to be the same ns as the secret provider class?
+<3> Specify the name of the secret provider class.
+
+.. Create the `Deployment` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f deployment.yaml
+----
+
+.Verification
+
+* Verify that you can access the secrets from AWS Secrets Manager in the pod volume mount:
+
+.. List the secrets in the pod mount:
++
+[source,terminal]
+----
+$ oc exec busybox-<hash> -n my-namespace -- ls /mnt/key-store/
+----
++
+.Example output
+[source,terminal]
+----
+tls-cert
+----
+
+.. View a secret in the pod mount:
++
+[source,terminal]
+----
+$ oc exec busybox-<hash> -n my-namespace -- cat /mnt/key-store/tls-cert
+----
++
+.Example output
+[source,terminal]
+----
+my-secret-value
+----
+// TODO: Is this okay example output, or should it be something more specific to our example?
```

Comment:
@xingxingxia @ropatil010 Does this basic example output look okay for the secret example we've used, or should this be something more detailed?

Reply:
Yes okay.